jme3-core/src/plugins/java/com/jme3/material/plugins/ShaderNodeLoaderDelegate.java

```diff
@@ -200,8 +202,26 @@ protected void readShaderNodeDefinition(List<Statement> statements, ShaderNodeDe
                             throw new MatParseException(e.getMessage(), statement1, e);
                         }
                     }
+                } else if (line.startsWith("Defines")) {
+                    for (final Statement sub : statement.getContents()) {
+                        try {
+                            final String[] values = sub.getLine().split("[ \\{]");
+                            shaderNodeDefinition.getDefines().add(values[0]);
+                        } catch (final RuntimeException e) {
+                            throw new MatParseException(e.getMessage(), sub, e);
+                        }
+                    }
+                } else if (line.startsWith("Imports")) {
+                    for (final Statement sub : statement.getContents()) {
+                        try {
+                            final String[] values = sub.getLine().split("[ \\{]");
+                            shaderNodeDefinition.getImports().add(values[0]);
+                        } catch (final RuntimeException e) {
+                            throw new MatParseException(e.getMessage(), sub, e);
+                        }
+                    }

```

Comment:
Why do we need this?
Defines and Imports can be defined in the shader node code... no need to do this in the shader node definition.

Reply:
What happened when you will have two shader nodes with the same import statements? But I agree that it isn't good to have "hardcoded(imports/defines)" additional information here, can I change it to using more generic solution?

Reply:
I need to check this part, I know there is a logic in the global shader system with this, but I can't test for now.